File: tests/indexing/parsing/lang/fixtures/python/Exceptions.py
def parse_int(text):
    value = 0
    try:
        value = int(text)
        value = value * 2
    except ValueError as e:
        value = -1
        text = str(e)
    finally:
        text = text.strip()
        value = value + len(text)
    return value


def element_at(xs, i, fallback):
    result = 0
    try:
        result = xs[i]
        result = result + i
    except IndexError:
        result = fallback
        fallback = fallback - 1
    except TypeError:
        result = 0
    return result

File: tests/indexing/parsing/lang/fixtures/php/Exceptions.php
<?php

function parseOrZero(string $text): int
{
    $value = 0;
    try {
        $value = intval($text);
        $value = $value * 2;
    } catch (InvalidArgumentException $e) {
        $value = 0;
        $text = $e->getMessage();
    } finally {
        $text = trim($text);
        $value = $value + strlen($text);
    }
    return $value;
}

function elementAt(array $xs, int $i, int $fallback): int
{
    $result = 0;
    try {
        $result = $xs[$i];
        $result = $result + $i;
    } catch (OutOfRangeException $e) {
        $result = $fallback;
        $fallback = $fallback - 1;
    } catch (TypeError $e) {
        $result = 0;
    }
    return $result;
}
